test/unit/liquidity_pool_id_test.js

```diff
@@ -0,0 +1,265 @@
+const assetA = new StellarBase.Asset(
+  'ARST',
+  'GB7TAYRUZGE6TVT7NHP5SMIZRNQA6PLM423EYISAOAP3MKYIQMVYP2JO'
+);
+const assetB = new StellarBase.Asset(
+  'USD',
+  'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+);
+const fee = StellarBase.LiquidityPoolFeeV18;
+
+describe('StellarBase#getLiquidityPoolId()', function() {
+  it('throws an error if the liquidity pool type is not `constant_product`', function() {
+    expect(() => StellarBase.getLiquidityPoolId()).to.throw(
+      /liquidityPoolType is invalid/
+    );
+
+    expect(() => StellarBase.getLiquidityPoolId(1)).to.throw(
+      /liquidityPoolType is invalid/
+    );
+
+    expect(() => StellarBase.getLiquidityPoolId('random_type')).to.throw(
+      /liquidityPoolType is invalid/
+    );
+  });
+
+  it('throws an error if assetA is invalid', function() {
+    expect(() =>
+      StellarBase.getLiquidityPoolId('constant_product', {})
+    ).to.throw(/assetA is invalid/);
+
+    expect(() =>
+      StellarBase.getLiquidityPoolId('constant_product', { assetA: 'random' })
+    ).to.throw(/assetA is invalid/);
+  });
+
+  it('throws an error if assetB is invalid', function() {
+    expect(() =>
+      StellarBase.getLiquidityPoolId('constant_product', {
+        assetA
+      })
+    ).to.throw(/assetB is invalid/);
+
+    expect(() =>
+      StellarBase.getLiquidityPoolId('constant_product', {
+        assetA,
+        assetB: 'random'
+      })
+    ).to.throw(/assetB is invalid/);
+  });
+
+  it('throws an error if fee is invalid', function() {
+    expect(() =>
+      StellarBase.getLiquidityPoolId('constant_product', {
+        assetA,
+        assetB
+      })
+    ).to.throw(/fee is invalid/);
+  });
+
+  it('returns poolId correctly', function() {
+    const poolId = StellarBase.getLiquidityPoolId('constant_product', {
+      assetA,
+      assetB,
+      fee
+    });
+
+    expect(poolId.toString('hex')).to.equal(
+      'dd7b1ab831c273310ddbec6f97870aa83c2fbd78ce22aded37ecbf4f3380fac7'
+    );
+  });
+
+  it('returns poolId correctly, based on stellar-core tests', function() {
+    // The tests below were copied from https://github.com/stellar/stellar-core/blob/c5f6349b240818f716617ca6e0f08d295a6fad9a/src/transactions/test/LiquidityPoolTradeTests.cpp#L430-L526
+    const issuer1 = StellarBase.StrKey.encodeEd25519PublicKey(
+      Buffer.from(
+        '0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef',
+        'hex'
+      )
+    );
+    const issuer2 = StellarBase.StrKey.encodeEd25519PublicKey(
+      Buffer.from(
+        'abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789',
+        'hex'
+      )
+    );
+
+    // NATIVE and ALPHANUM4 (short and full length)
```

Comment:
It would be more descriptive if these comments were test names, e.g.

```js
it("works when mixing native and alpha4 assets",  ...)
```

Thoughts?

Reply:
Good idea! Done in https://github.com/stellar/js-stellar-base/pull/454/commits/b0060f2c510cf51bb1048265a326d138aab67c6c